### PR TITLE
openusd_internal: patch reduction progress

### PR DESCRIPTION
**Pull request by rpoyner-tri**
* enable upgrade.py to work vs OpenUSD dev branch
* steal a symbol hiding trick from patches into the build system

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)

### CODE REVIEW

**Review by rpoyner-tri — COMMENTED** [review]
+@jwnimmer-tri for feature review at your leisure.

This set of patches is what I used to get drake to build against the OpenUsd dev branch, which is where upstream patches must be tested and submitted.

Their dev is quite a distance from our reliance on a numbered release, and it pulls in more sub-libraries of OpenUsd. One consequence is we pull a library that invalidates the assumptions of the previously implemented cmake-file scraping code. We can't count on all of the literal file names, etc., living lexically within the `pxr_library()` invocation any more. Instead, this code uses the cmake interpreter and specially stubbed functions to print out the arguments of interest, after variable expansion. There are still limitations (cmake include statements, the need to stub out whatever is called from CMakeLists.txt files, etc.), but it is a bit more flexible than the prior scheme, and probably could be expanded.

Note that none of this is needed by our currently integrated revision of OpenUsd, so this PR could hang around for a while before we need to merge it.

FWIW, the currently submitted upstream patches are these:
https://github.com/PixarAnimationStudios/OpenUSD/pull/3282
https://github.com/PixarAnimationStudios/OpenUSD/pull/3281
https://github.com/PixarAnimationStudios/OpenUSD/pull/3280
https://github.com/PixarAnimationStudios/OpenUSD/pull/3278

They are slightly different in form than the current drake patches, which are a little rougher and less upstream-friendly.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6MT2fs4jdqQEL6Rscl:bbe6kq4)* status: LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

**Review by jwnimmer-tri — COMMENTED** [review]
LGTM but a few questions.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6NDPPxB0P_sEtUjdvD:bmvaiy6)* status: 7 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/defs.bzl` line 58 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6N808_A9oGyJMiKljg:-O6N808_A9oGyJMiKljh:bigm56k) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/defs.bzl#L58)):*
> steal a symbol hiding trick from patches into the build system

I'm not sure _why_ this is change is being proposed.  We have `patches/nanocolor_namespace.patch` whose last hunk already seems to set this unconditionally (and that patch is not rm'd as part of this PR), so now we're setting it in two places?

Bigger picture, in my view generally this `defs.bzl` file is intended to provide for something analogous to the upstream build system (without Drake-specific customizations).  The Drake customizations generally should live in patches, not here.  But maybe there is a reason to change that posture now?
___
*[`tools/workspace/openusd_internal/upgrade.py` line 40 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NCoHc22kx4ssQs5qD:-O6NCoHc22kx4ssQs5qE:bnmmy8b) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L40)):*
> ```Python
> # argument parsing for pxr_library() must match that file. The remaining
> # functions are just no-op stubs as required by the CMakeLists.txt files found
> # in the subdirs listed above.
> ```

How do you feel about scraping this macro instead of copy-pasting it?  I imagine that scraping `Public.cmake ` from the `function(pxr_library NAME)` needle up through the next `)` after the `cmake_parse_arguments(` needle would be fairly durable?

I suggest that both because it would plausibly adapt automatically when upstream changes their cmake goo in a meaningful way but one that still matches our needles, and because then we don't need to deal with propagating the upstream copyright text into here.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 88 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NC-Et5tPC3NK5anp0:-O6NC-Et5tPC3NK5anp1:b5al5zy) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L88)):*
> ```Python
> endfunction()
> 
> function(pxr_build_test)
> ```

BTW I imagine that I would be happier with the readability here if we had an acute list of ...

```py
CMAKE_IGNORED_FUNCTIONS = [
    "pxr_build_test",
    "pxr_test_scripts",
    ....
]
```

... and then generated the CMake-ignore-this-function boilerplate with a recipe down brlow.  It would help us focus on the important point of this content, which is the stuff up above.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 113 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NDxCQ3ew1GbbvYWDB:-O6NDxCR4SNnT7WUG0ii:blh0b0a) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L113)):*
> ```Python
>     pxr_library() call in the given cmake source code string."""
>     script = CMAKE_CODE_STUB + cmake
>     with tempfile.NamedTemporaryFile() as tmpf:
> ```

nit `tmpf` invalid abbreviation
___
*[`tools/workspace/openusd_internal/upgrade.py` line 116 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NE-NB8sX1ifn3QTDj:-O6NE-NB8sX1ifn3QTDk:b-fdfv5u) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L116)):*
> ```Python
>         tmpf.write(script.encode("utf8"))
>         tmpf.flush()
>         cmd = ["cmake", "-P", tmpf.name]
> ```

nit `cmd` invalid abbreviation.  (Upstream calls is `args`, which would be OK here is you like.  Or `command` is fine too.)
___
*[`tools/workspace/openusd_internal/upgrade.py` line 120 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NADo_3UnU42fArsL2:-O6NADo_3UnU42fArsL3:b3i8ar1) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L120)):*
> ```Python
>             cmd, stderr=subprocess.STDOUT).decode("utf8")
> 
>     result = collections.OrderedDict()
> ```

nit `OrderedDict` is vestigial fluff from early-Python days.  A vanilla `dict()` is already ordered.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 125 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NBSVdD5VkMcY5liBT:-O6NBSVdD5VkMcY5liBU:b-foet5k) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L125)):*
> ```Python
>             continue
>         kv = line.split(" = ")
>         if len(kv) == 1:
> ```

When would `len(kv) == 1` ever be true?  (This seems like dead code.)  Above, we already did a `continue` when the line did not contain ` = `.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 7 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/defs.bzl` line 58 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6N808_A9oGyJMiKljg:-O6b-iJt3NYuC7_nodig:b-7jxqt7) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/defs.bzl#L58)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

> steal a symbol hiding trick from patches into the build system

I'm not sure _why_ this is change is being proposed.  We have `patches/nanocolor_namespace.patch` whose last hunk already seems to set this unconditionally (and that patch is not rm'd as part of this PR), so now we're setting it in two places?

Bigger picture, in my view generally this `defs.bzl` file is intended to provide for something analogous to the upstream build system (without Drake-specific customizations).  The Drake customizations generally should live in patches, not here.  But maybe there is a reason to change that posture now?
</blockquote></details>

I'll withdraw this shortly. It's probably fine for something like this line to remain a patch. It would be convenient, however for the `defs` code to take preprocessor definitions (not cmake definitions) as input.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 7 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/defs.bzl` line 58 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6N808_A9oGyJMiKljg:-O6c3qynAIIQwFGpucRR:brbc6z) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/defs.bzl#L58)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

I'll withdraw this shortly. It's probably fine for something like this line to remain a patch. It would be convenient, however for the `defs` code to take preprocessor definitions (not cmake definitions) as input.
</blockquote></details>

Around line 30 up above we do have some project-wide definitions already (`defines = ...`), but those are only for things that are global settings.

For settings local to just one library, my premise has been that the CMakeLists are the source of truth, and if we need a different truth then we patch the CMake spelling.  Or sometimes, settings are in the C layer instead of the CMake layer, such as in preprocessor `#ifndef` defaults, in which case the source of truth is the C code so we patch the C code.

In our [VTK build](https://github.com/RobotLocomotion/drake/blob/master/tools/workspace/vtk_internal/settings.bzl) I did take the route of using a `settings.bzl`  file as the source of truth for build options, separate from the code, but it's a lot of work to maintain.  Therefore, for simpler projects (like USD), keeping the truth directly inline in the source tree is likely to be the most durable option.

That's just practical tactics though, so if we have evidence that a different approach would be a net win, we can certainly consider it.  I just don't know what problem you're trying to solve, yet.  My mental model of rebasing our patches in preparation for upstreaming them is _all_ of our patches are cherry-picked into their tree and then rebased and cleaned up, all as stacked curated commits on top of their master.  If the challenge here was trying to rebase the patches one by one without the others already in evidence, then we shouldn't expect Drake to ever be OK with just a subset of the patches.  (Maybe I'm way off base.  Just guessing.)

**Review by rpoyner-tri — COMMENTED** [review]
Progress. Still working on the cmake scraping and scripting.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6gMgdC9Nl7XodZ2krX:bp1mrej)* status: 4 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/defs.bzl` line 58 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6N808_A9oGyJMiKljg:-O6gGgM06DmWSpHTMTR8:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/defs.bzl#L58)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Around line 30 up above we do have some project-wide definitions already (`defines = ...`), but those are only for things that are global settings.

For settings local to just one library, my premise has been that the CMakeLists are the source of truth, and if we need a different truth then we patch the CMake spelling.  Or sometimes, settings are in the C layer instead of the CMake layer, such as in preprocessor `#ifndef` defaults, in which case the source of truth is the C code so we patch the C code.

In our [VTK build](https://github.com/RobotLocomotion/drake/blob/master/tools/workspace/vtk_internal/settings.bzl) I did take the route of using a `settings.bzl`  file as the source of truth for build options, separate from the code, but it's a lot of work to maintain.  Therefore, for simpler projects (like USD), keeping the truth directly inline in the source tree is likely to be the most durable option.

That's just practical tactics though, so if we have evidence that a different approach would be a net win, we can certainly consider it.  I just don't know what problem you're trying to solve, yet.  My mental model of rebasing our patches in preparation for upstreaming them is _all_ of our patches are cherry-picked into their tree and then rebased and cleaned up, all as stacked curated commits on top of their master.  If the challenge here was trying to rebase the patches one by one without the others already in evidence, then we shouldn't expect Drake to ever be OK with just a subset of the patches.  (Maybe I'm way off base.  Just guessing.)
</blockquote></details>

Done.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

**Review by rpoyner-tri — COMMENTED** [review]
Mostly done. Still needs more testing and probably pointy wizard scrutiny of regex usage.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6hROoo29Uy1n21CFhi:bhmbicz)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/upgrade.py` line 40 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NCoHc22kx4ssQs5qD:-O6hREzxDAnfXzfzd3Zj:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L40)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

How do you feel about scraping this macro instead of copy-pasting it?  I imagine that scraping `Public.cmake ` from the `function(pxr_library NAME)` needle up through the next `)` after the `cmake_parse_arguments(` needle would be fairly durable?

I suggest that both because it would plausibly adapt automatically when upstream changes their cmake goo in a meaningful way but one that still matches our needles, and because then we don't need to deal with propagating the upstream copyright text into here.
</blockquote></details>

Done.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 88 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6NC-Et5tPC3NK5anp0:-O6hRH4-9oWxCoCjEVbI:b5negzl) ([raw file](https://github.com/RobotLocomotion/drake/blob/4da034c480a6a7c7d9d1e567ee1ff80db2e05423/tools/workspace/openusd_internal/upgrade.py#L88)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW I imagine that I would be happier with the readability here if we had an acute list of ...

```py
CMAKE_IGNORED_FUNCTIONS = [
    "pxr_build_test",
    "pxr_test_scripts",
    ....
]
```

... and then generated the CMake-ignore-this-function boilerplate with a recipe down brlow.  It would help us focus on the important point of this content, which is the stuff up above.
</blockquote></details>

the list is now built by scraping the macro file; PTAL.

**Review by rpoyner-tri — COMMENTED** [review]
Fixed lint stuff and retested builds vs both current drake dep and upstream dev branch.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6qXAw166tTUQs4mJvs:bhna52k)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

**Review by jwnimmer-tri — COMMENTED** [review]
:lgtm: feature.

> Reviewed 2 of 3 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-:-O6vFhlm7WQPtoQsKF_l:b-n37teh)* status: 4 unresolved discussions, needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/package.BUILD.bazel` line 21 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6i-LBi47LGr6572PlG:-O6i-LBi47LGr6572PlH:b-7vvd7g) ([raw file](https://github.com/RobotLocomotion/drake/blob/7d1bd39949f639ee893899b144e3deaf92269ff5/tools/workspace/openusd_internal/package.BUILD.bazel#L21)):*
> ```
> filegroup(
>     name = "cmake_macros",
>     srcs = glob(["cmake/macros/Public.cmake"]),
> ```

nit If this is a `glob` in order to allow the file to be nominally absent, then we should say `allow_empty = True` specifically to communicate that intention.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 40 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6vFc616pveaJrxw07e:-O6vFc627nYAxJFmrQna:bom2yiq) ([raw file](https://github.com/RobotLocomotion/drake/blob/532c8e1e9fe21b907d0658abfc735d70e2610e4b/tools/workspace/openusd_internal/upgrade.py#L40)):*
> ```Python
> # argument parsing for pxr_library() must match that file. The remaining
> # functions are just no-op stubs as required by the CMakeLists.txt files found
> # in the subdirs listed above.
> ```

nit This comment paragraph is stale.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 70 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6vFQUWDRieg25zXFZ-:-O6vFQUWDRieg25zXFZ0:bb7phvt) ([raw file](https://github.com/RobotLocomotion/drake/blob/532c8e1e9fe21b907d0658abfc735d70e2610e4b/tools/workspace/openusd_internal/upgrade.py#L70)):*
> ```Python
> def _compose_cmake_prelude():
>     """Scrape the contents of the cmake/macros/Public.cmake file from OpenUSD.
>     Compose a prelude of redefined (mostly stubbed-out) function to use when
> ```

typo



_Suggestion:_
```Python
functions
```
___
*[`tools/workspace/openusd_internal/upgrade.py` line 97 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6vEXrn-Vuyy9J2Pbg4:-O6vEXrn-Vuyy9J2Pbg5:b-bmo117) ([raw file](https://github.com/RobotLocomotion/drake/blob/532c8e1e9fe21b907d0658abfc735d70e2610e4b/tools/workspace/openusd_internal/upgrade.py#L97)):*
> ```Python
> 
> 
> def _interpret(prelude: str, cmake: str):
> ```

Having two arguments here doesn't make sense.

Either this function is a low-level helper in which case it takes a simple `cmake: str` as its sole argument and just runs it (in which case the caller should pre-join the `prelude` and `cmake` strings), or else (the more likely option) this function is acutely a NAME=VALUE extractor, in which case the prelude is _not_ actually customizable and should not be an argument.

In other words, it seems to me like the `prelude = _compose_cmake_prelude()` should happen inside this function, or else I've misunderstood the code and something else needs to be cleared up.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 4 unresolved discussions, needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/package.BUILD.bazel` line 21 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6i-LBi47LGr6572PlG:-O6vH0FPBnxbFhpeoxZ1:b-a9tbi) ([raw file](https://github.com/RobotLocomotion/drake/blob/7d1bd39949f639ee893899b144e3deaf92269ff5/tools/workspace/openusd_internal/package.BUILD.bazel#L21)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

nit If this is a `glob` in order to allow the file to be nominally absent, then we should say `allow_empty = True` specifically to communicate that intention.
</blockquote></details>

:facepalm: it doesn't even need to be a glob.
___
*[`tools/workspace/openusd_internal/upgrade.py` line 97 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6vEXrn-Vuyy9J2Pbg4:-O6vHCLu00tsTK8-To3g:but2gtk) ([raw file](https://github.com/RobotLocomotion/drake/blob/532c8e1e9fe21b907d0658abfc735d70e2610e4b/tools/workspace/openusd_internal/upgrade.py#L97)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Having two arguments here doesn't make sense.

Either this function is a low-level helper in which case it takes a simple `cmake: str` as its sole argument and just runs it (in which case the caller should pre-join the `prelude` and `cmake` strings), or else (the more likely option) this function is acutely a NAME=VALUE extractor, in which case the prelude is _not_ actually customizable and should not be an argument.

In other words, it seems to me like the `prelude = _compose_cmake_prelude()` should happen inside this function, or else I've misunderstood the code and something else needs to be cleared up.
</blockquote></details>

We don't really need to rebuild the prelude 20 times. Other options are to make prelude a global, or to scoop many of the free functions into a class, so we can manage state properly.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 4 unresolved discussions, needs at least two assigned reviewers

___
*[`tools/workspace/openusd_internal/upgrade.py` line 97 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O6vEXrn-Vuyy9J2Pbg4:-O6vIpnL1RymLK6lK_mG:b-xdrecs) ([raw file](https://github.com/RobotLocomotion/drake/blob/532c8e1e9fe21b907d0658abfc735d70e2610e4b/tools/workspace/openusd_internal/upgrade.py#L97)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

We don't really need to rebuild the prelude 20 times. Other options are to make prelude a global, or to scoop many of the free functions into a class, so we can manage state properly.
</blockquote></details>

Ah, it's just for performance?  Use `@functools.cache` on the prelude calculator and then we can lose the extra plumbing.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: needs at least two assigned reviewers

**Comment by jwnimmer-tri** [comment]
+@ggould-tri for platform review per schedule, please.

The context here is that `upgrade.py` is a maintainer tool run by hand by build system maintainers, when we need to switch our pin of OpenUSD to a newer/different version.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 4 unresolved discussions

___
*[`tools/workspace/openusd_internal/upgrade.py` line 90 at r5](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O7-CxQL77PLpP4DW_wy:-O7-VKEy9-s4O52z6cjs:bxcbgo9) ([raw file](https://github.com/RobotLocomotion/drake/blob/cc544e6880e74017151dcce3ed967ed2645a7980/tools/workspace/openusd_internal/upgrade.py#L90)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

nit: `TODO` should have username or issue number.
</blockquote></details>

actually, it's done. will remove
___
*[`tools/workspace/openusd_internal/upgrade.py` line 107 at r5](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O7-DP5K6PoEzVxk_54T:-O7-V_A1EsrRWsAwtOwZ:b-udyaa5) ([raw file](https://github.com/RobotLocomotion/drake/blob/cc544e6880e74017151dcce3ed967ed2645a7980/tools/workspace/openusd_internal/upgrade.py#L107)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

minor:  Consider setting `prefix` and/or `suffix` to clarify the file's purpose and semantics.
</blockquote></details>

actually, it's neither. it's the combined whole script. will rename

**Review by ggould-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: 4 unresolved discussions

___
*[`tools/workspace/openusd_internal/upgrade.py` line 107 at r5](https://reviewable.io/reviews/RobotLocomotion/drake/21886#-O7-DP5K6PoEzVxk_54T:-O703mYM7oLJMwzxpEQa:b-t2ozvt) ([raw file](https://github.com/RobotLocomotion/drake/blob/cc544e6880e74017151dcce3ed967ed2645a7980/tools/workspace/openusd_internal/upgrade.py#L107)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

actually, it's neither. it's the combined whole script. will rename
</blockquote></details>

I meant the `prefix` and `suffix` kwargs to `NamedTemporaryFile`; sorry if I was unclear.

**Review by ggould-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: <span>:shipit:</span> complete! all discussions resolved, LGTM from assignees ggould-tri(platform),jwnimmer-tri(platform)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/21886)* status: <span>:shipit:</span> complete! all discussions resolved, LGTM from assignees ggould-tri(platform),jwnimmer-tri(platform)
